### ISSUE TITLE
`pext/yotta:latest` image type has been deprecated and stopped working with latest docker

### ISSUE THREAD

**Issue by microbit-carlos** [microsoft/pxt-microbit#5721]
Trying to build Makecode with Docker v26+ fails, as trying to pull the `pext/yotta:latest` image (for the V2 CODAL build) throws the following error:

```
$ docker pull pext/yotta:latest
latest: Pulling from pext/yotta
[DEPRECATION NOTICE] Docker Image Format v1 and Docker Image manifest version 2, schema 1 support is disabled by default and will be removed in an upcoming release. Suggest the author of docker.io/pext/yotta:latest to upgrade the image to the OCI Format or Docker Image manifest v2, schema 2. More information at https://docs.docker.com/go/deprecated-image-specs/
```
https://github.com/microsoft/pxt-microbit/blob/bd4bda19f33ba30ea61e3620c8fa069b8306d07e/pxtarget.json#L204-L207

This docker image type was deprecated in Docker v19, it has been disabled already in v26, and v27 (which was release a couple of weeks ago) has completely and permanently removed the functionality:
https://docs.docker.com/engine/deprecated/#pushing-and-pulling-with-image-manifest-v2-schema-1

The GH Actions runner images (which we use in the CODAL CI to test changes with a MakeCode build) currently use Docker v26, which needs a workaround to enable this image type, but it won't be long before they upgrade to v27 and things will stop working if there isn't a way to downgrade the docker version from the images.

**Comment by riknoll** [comment]
we're working on publishing these images to Microsoft's public container registry. unfortunately, due to licensing issues, i can't update the existing images in the pext dockerhub repo

**Comment by microbit-carlos** [comment]
Ah, good to know, thanks for the update @riknoll.
Will pushing the `pext/yotta:latest` image to the new registry using a modern docker version update the image manifest type? Or will it need to be rebuilt?

If it needs to be rebuilt it might be a good opportunity to upgrade the image, as `pext/yotta:latest` is using GCC v6, and something a bit more modern like v10.3 might improve build sizes (just an assumption based on experience, but of course we won't know until we try it and compare).

**Comment by riknoll** [comment]
@microbit-carlos I'll be rebuilding the images before I push them, so no worries there! I've actually already been updating all of the dockerfiles used by our compile service backend, just need somewhere to put them for public consumption.

totally down for experimenting with different gcc versions!

**Comment by abchatra** [comment]
If this is still an issue reopen.

**Comment by microbit-carlos** [comment]
This is still an issue, and containerd v2.1 is already released, so at some point Docker will update its internal containerd and this image will be completely unrunnable.

```bash
$ docker pull pext/yotta:latest
Error response from daemon: failed to get converter for "docker.io/pext/yotta:latest":
Pulling Schema 1 images have been deprecated and disabled by default since
containerd v2.0. As a workaround you may set an environment variable
`CONTAINERD_ENABLE_DEPRECATED_PULL_SCHEMA_1_IMAGE=1`, but this will
be completely removed in containerd v2.1.
```

And with the current Docker (well, the version I installed in a new laptop a month ago) I cannot run the image directly anymore. Looks like I need to set up the `CONTAINERD_ENABLE_DEPRECATED_PULL_SCHEMA_1_IMAGE` variable in the docker daemon (is not enough to run the docker command with that env variable) and so far struggling to set that up in macOS.

**Comment by microbit-carlos** [comment]
Based on the Docker release notes (https://docs.docker.com/engine/deprecated/#pushing-and-pulling-with-image-manifest-v2-schema-1) it sounds like even if the current Docker release is not using containerd v2.1, it has already removed support for this flag, so I think this image is already unrunnable with the latest release:

> Support for the `DOCKER_ENABLE_DEPRECATED_PULL_SCHEMA_1_IMAGE` environment variable will be removed in Docker v27.0 after which this functionality is removed permanently.

**Comment by fabianhugo** [comment]
I can confirm, that pext/yotta:latest is currently unrunnable via docker.

**Comment by bsiever** [comment]
Is there currently any way to do local build via docker (e.g., disable dal or a setting or something)?

**Comment by microbit-carlos** [comment]
To build only for CODAL you can use this environmental variable when running the pxt cli commands: `PXT_COMPILE_SWITCHES=csv---mbcodal`

**Comment by bsiever** [comment]
Thanks @microbit-carlos , but I'm still unable to do local builds (via Docker) due to the manifest.  I'm on macOS; Docker Desktop 4.74.0. Is there something I'm doing wrong or a fix?

Error is:
``` 
"Unable to find image 'pext/yotta:latest' locally
docker: Error response from daemon: not implemented: media type "application/vnd.docker.distribution.manifest.v"
...
```

**Comment by microbit-carlos** [comment]
Right, that's the current issue with the docker images, `pext/yotta:latest` is used for CODAL and `pext/yotta:gcc5` for DAL. So building for DAL with `PXT_COMPILE_SWITCHES=csv---mbdal` should work (if you don't hit GH download rate limits), and building for CODAL fails:

```bash
$ docker run --rm pext/yotta:latest yotta --version
Unable to find image 'pext/yotta:latest' locally
docker: Error response from daemon: not implemented: media type "application/vnd.docker.distribution.manifest.v1+prettyjws" is no longer supported since containerd v2.1, please rebuild the image as "application/vnd.docker.distribution.manifest.v2+json" or "application/vnd.oci.image.manifest.v1+json"

Run 'docker run --help' for more information
```
```bash
$ docker run --rm pext/yotta:gcc5 yotta --version
0.20.5
```

https://github.com/microsoft/pxt-microbit/blob/4e48178f5c15cc46afdd423bdff3ec140a284d54/pxtarget.json#L175-L180

https://github.com/microsoft/pxt-microbit/blob/4e48178f5c15cc46afdd423bdff3ec140a284d54/pxtarget.json#L206-L210

My only work around at the moment is to build with local tooling:

```
PXT_COMPILE_SWITCHES=csv---mbcodal PXT_NODOCKER=1 PXT_FORCE_LOCAL=1 PXT_RUNTIME_DEV=1 npx pxt serve --local
```
